lib/datetime/helpers.ex

```diff
@@ -145,4 +145,28 @@ defmodule Timex.DateTime.Helpers do
       new_p
     end
   end
+
+  def sort_options(options) when is_list(options) do
```

Comment:
I would instead do something like this:

```elixir
@set_option_priority %{
  year: 1,
  month: 2,
  day: 3,
  hour: 4,
  minute: 5,
  second: 6,
}

def sort_options(options) when is_list(options) do
  Enum.sort_by(options, fn {k, _} -> Map.get(@set_option_priority, k, 99) end)
end
```

This is a lot easier to maintain and a lot more readable, and ensures that other options are always sorted _after_ the explicitly prioritized attributes. Really we should define the priority for all valid options, but these will do for now.